Code/Editor/Platform/Windows/Editor/Core/QtEditorApplication_windows.cpp

```diff
@@ -120,7 +120,19 @@ namespace Editor
             }
             else if (msg->message == WM_KEYDOWN || msg->message == WM_KEYUP || msg->message == WM_CHAR)
             {
+#if defined(CARBONATED)
+                // Process raw WM_CHAR event sending keycode to a LyShine UiTextInputComponent in Editor game mode
+                if (filterInputEvents && msg->message == WM_CHAR)
+                {
+                   AzFramework::RawInputNotificationBusWindows::Broadcast(
+                        &AzFramework::RawInputNotificationsWindows::OnRawInputCodeUnitUTF16Event, msg->wParam);
+                    return true;
+                }
+                // Returning true filters out all keyboard events in Editor game mode, thus preventing input to a LyShine UiTextInputComponent
+                return false;
+#else
                 return filterInputEvents;
+#endif // defined(CARBONATED)
             }
```

Comment:
This should reworked a little to be less confusing.  Also, if WM_KEYDOWN, WM_KEYUP, it will always return false instead of "filterInputEvents".

I suggest the following:

```
 else if (msg->message == WM_CHAR)
{
    // Process raw WM_CHAR event sending keycode to a LyShine UiTextInputComponent in Editor game mode
    if (filterInputEvents)
    {
        AzFramework::RawInputNotificationBusWindows::Broadcast(&AzFramework::RawInputNotificationsWindows::OnRawInputCodeUnitUTF16Event, msg->wParam);
        return true;
    }
    // Returning true filters out all keyboard events in Editor game mode, thus preventing input to a LyShine UiTextInputComponent
    return false;
}
else if(msg->message == WM_KEYDOWN || msg->message == WM_KEYUP)
{
    return filterInputEvents;
}
```

Reply:
(with CARBONATED guards)

Reply:
Returning "false" to `DefWndProc` on `WM_KEYDOWN` and `WM_KEYUP` is critical to a Win32/64 character input, as `WM_CHAR` is only generated in `DefWndProc` when `WM_KEYDOWN` and `WM_KEYUP` are bypassed unhandled, considering current keyboard layout and "Caps lock" and "Shift" kets state.
Proposed above code does not work, code in PR does work.

Reply:
@carb-lloyd 
Restructured code of key-input handler for Editor in-game mode, hope this variant is better readable.
Sorry for delay, switching branches and recompiling for checks takes time.